### PR TITLE
Add new allocated federal/nonfederal disbursements datatable

### LINKED ISSUE #5651: Develop H4 datatable skeleton
**What we're after:**
While back-end work is still ongoing, front-end can begin to build out the skeleton of the H4 datatable. 

@JonellaCulmer will update this ticket with the design mockups completed in 21.1.

**Action items:**
- [ ] TBD

**Completion criteria:**
- [ ] Skeleton complete and ready for back-end work to be incorporated.

### LINKED ISSUE #5652: Conduct UAT for H4 datatable and associated pages
**What we're after:**
Conduct UAT in week 1 of the sprint to allow for front-end updates in week 2. 

**ITEMS TO CHECK:**

Design: (datatable, committee profile pages, browse data page)
- [ ] Overall appearance
- [ ] UI/UX patterns utilized correctly
- [ ] Matches design mockups/functionality, as appropriate
- [ ] Properly linked to commitee profile pages
- [ ] Browse data page(s) are accurate

Content: (datatable, browse data page(s))
- [ ] Language

Front-end (datatable, committee profile pages)
- [ ] Code
- [ ] Identify any potential bugs
- [ ] Correct linkages

Back-end
- [ ] Data is appearing as expected

### Completion criteria
- [ ] Datatable has been checked by all subgroups and approved for launch

### LINKED ISSUE #5654: Prep for launch of H4 feature work
**What we're after:**
We are preparing to launch the H4 feature work which includes:
- Datatable
- Committee profile page linkages
- Browse data page(s) additions

UAT happening here: https://github.com/fecgov/fec-cms/issues/5652

**Completion criteria:**
- [x] Updates to the H4 feature work has been completed based on UAT feedback

### PR DESCRIPTION
**Pull request by patphongs**
## Summary

- Resolves #5651 
- Resolves #5652 
- Resolves #5654 

### Required reviewers

- UX
- front-end
- back-end
- content
- SMEs?

## Impacted areas of the application

- Created a feature flag (`h4_allocated_disbursements`) and wrapped the new H4 work in it
- Repo-wide: "Non-federal transfers" was changed to "**Nonfederal** transfers"
- Repo-wide: "Allocated operating expenditures - federal" changed to "Allocated operating expenditures - federal **share**"
- Repo-wide: "Allocated operating expenditures - non-federal" changed to "Allocated operating expenditures - **nonfederal** share"
- Repo-wide: "Non-federal transfers" changed to "**Nonfederal** transfers"
- Repo-wide: "Total non-federal transfers" changed to "Total **nonfederal** transfers"
- Repo-wide: "Non-federal operating expenditures" changed to "**Nonfederal** operating expenditures"
- Details panels: changed `{{city}}, {{state}}, {{zip}}` to `{{city}}, {{state}} {{zip}}` (debts, disbursements, loans, receipts)
- [Spending page](http://127.0.0.1:8000/data/browse-data/?tab=spending) - added option for "Allocated federal/nonfederal disbursements", which links to…
- The [Allocated federal/nonfederal disbursements](http://127.0.0.1:8000/data/allocated-federal-nonfederal-disbursements) page/tables and filters were created
- Text filters can now have a different type than `"text"` but it's only been implemented in this datatable's ZIP Code filter

## Screenshots

Table
image

Details panel open
image

Committee summary page
image

Spending page
image

## Related PRs

None?

## How to test

- Check the [Known issues](#known-issues)
- pull the branch
- `npm i`
- `npm run build`
- `./manage.py runserver`
- Check the [Allocated federal/nonfederal disbursements table](http://127.0.0.1:8000/data/allocated-federal-nonfederal-disbursements/)
   - review the details panels
   - try to break the filters
- Click a committee to review its overview page
- Check the [spending](http://127.0.0.1:8000/data/browse-data/?tab=spending) page
   - "Allocated federal/nonfederal disbursements" should be listed between "Party coordinated expenditures" and  "Electioneering communications"
- Review the other items listed in [Impacted areas](#impacted-areas-of-the-application)

## Known issues:
- The date column will often be blank
- City filters need uppercase city names. Back-end is working on a fix
- There's a feature flag for this. Localhost should always show it

### CODE REVIEW

**Review by cnlucas — APPROVED** [review]
This is mostly working well for me. We need to implement a solution on the back-end for hiding the null values as I think that there's a bug with that filter. The other bug has been fixed. I tested exporting the data and the impacted areas and couldn't find any other bugs. 

I have started a list of the current filter bugs on the [back end follow up ticket. ](https://github.com/fecgov/openFEC/issues/5364)


Searching by recipient isn't working for me right now.

**Comment by cnlucas** [comment]
Sort_hide_null is working for me. I would recommend adding sort_hide_null=true and then having it function like our disbursements datatable which automatically adds the current cycle:

Spending landing page
https://www.fec.gov/data/disbursements/?data_type=processed&two_year_transaction_period=2024&min_date=01%2F01%2F2023&max_date=12%2F31%2F2024 

Removing the filters
https://www.fec.gov/data/disbursements/?data_type=processed

**Review by cnlucas — APPROVED** [review]
LGTM thanks @rfultz

**Review by johnnyporkchops — APPROVED** [review]

File: fec/data/templates/partials/allocated-federal-nonfederal-disbursements-filter.jinja
[the remark this answers is not in the document]
Comment (rfultz) [reply fec/data/templates/partials/allocated-federal-nonfederal-disbursements-filter.jinja:1]: @cnlucas just following up. Has this been addressed?

[the remark this answers is not in the document]
Comment (rfultz) [reply fec/data/templates/partials/allocated-federal-nonfederal-disbursements-filter.jinja:1]: @cnlucas just following up. Has this been addressed?

[the remark this answers is not in the document]
Comment (cnlucas) [reply fec/data/templates/partials/allocated-federal-nonfederal-disbursements-filter.jinja:1]: @rfultz yep!

**Review by JonellaCulmer — APPROVED** [review]
Everything looks good and also passed stakeholder reviews.

**Pull request merged**